src/SFA.DAS.Funding.SystemAcceptanceTests/StepDefinitions/RecalculateEarningsAndPaymentsStepDefinitions.cs

```diff
@@ -209,7 +213,7 @@ public void AgreedPriceOnTheEarningsEntityIsUpdated(decimal agreedPrice)
 
             var apprenticeshipEntity = apiClient.GetEarningsEntityModel();
 
-            Assert.AreEqual(agreedPrice, apprenticeshipEntity.Model.AgreedPrice);
+            Assert.AreEqual(agreedPrice, apprenticeshipEntity.Model.ApprenticeshipEpisodes.First().Prices.OrderByDescending(x => x.ActualStartDate).First().AgreedPrice);
```

Comment:
I think we maybe want to update them all to match this one and get the last one based on the actual start date rather than the last one in the unordered list? Thoughts? If we are all in agreement I am happy to propagate this change through or leave it for you Umar if you would prefer to do it yourself?

Reply:
Solid Linq performance knowledge, @ShakilUmar let me know if you want me to implement this change or if you want to do it.